dashboard/lib/announcements.rb

```diff
@@ -24,7 +24,9 @@ def self.get_announcement_for_page(page)
   end
 
   def self.load_announcements
-    unless @@load_error || @@loaded
+    # Reloads JSON file with announcement data on each page load
+    # in non-production environments
+    unless (@@load_error || @@loaded) && Rails.env.production?
```

Comment:
Should we maybe attempt to reload if there was a `@@load_error`?

Reply:
You mean for our production environment? I think either the JSON is bad / doesn't exist / etc. until the next DTP, at which point `@@load_error` would be reset?

Reply:
More for our non-production environment... it might be an edge case, but I was thinking that if there was a problem loading the file one time, it might be worth trying to load again next time, and not skipping just because we'd seen the error before.  (Assuming I'm reading this right.. `unless` always throws me for a loop.)

Reply:
I _think_ the code within this `unless` will run in any non-production environment regardless of the value of `@@load_error` since the second part of the condition will be false?

Reply:
Oh you're right.  My bad.  Seems good then.